### PR TITLE
Usage of a helper for trace name when preparing trace data

### PR DESCRIPTION
**Pull request by swapster**
In continue of #541.

Now there is a `getTraceName()` method that can deal with a "root spans" with missing references. But it doesn't use when traces data transforms for a "Search" and "Compare" pages.
Seems like it should be using here.

### CODE REVIEW

**Comment by bogdanov1609** [comment]
Any updates here?

**Comment by swapster** [comment]
> Can there be a unit test?

Added some [tests](https://github.com/jaegertracing/jaeger-ui/pull/544/commits/50bb1911345fcdaf0758004f1de23d83592b1f75).

File: packages/jaeger-ui/src/model/transform-trace-data.test.js [hunk 1]
```diff
@@ -53,3 +53,191 @@ describe('deduplicateTags()', () => {
...
+      {
+        key: 'http.method',
+        type: 'string',
+        value: 'get',
+      },
+      {
+        key: 'http.status_code',
+        type: 'float64',
+        value: 200,
+      },
```
Comment (yurishkuro) [inline packages/jaeger-ui/src/model/transform-trace-data.test.js:191]: Nit: are these tags relevant to the functionality being tested? They don't look like it. Could you please trim the fixtures to minimally required data?

Comment (swapster) [reply packages/jaeger-ui/src/model/transform-trace-data.test.js]: Sounds reasonable, thanks :)
[Removed](https://github.com/jaegertracing/jaeger-ui/pull/544/commits/246db5249943e2e0bccfff64bdff967c78dc3e0a) unnecessary tags.
